packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -239,13 +239,17 @@ export class DeploymentManager {
   };
 
   private waitForIndices = async (stackParams: DeploymentMachineOp) => {
-    if (stackParams.tableNames.length) console.log('\nWaiting for DynamoDB table indices to be ready');
+    if (stackParams.tableNames.length) {
+      console.log('\nWaiting for DynamoDB table indices to be ready');
+      // cfn is async to ddb gsi creation the api can return true before the gsi creation starts
+      await new Promise(res => setTimeout(res, 2000));
```

Comment:
This reduces the chance of encountering the race condition, but doesn't remove it, right?

Reply:
Correct, though afaik there is no method to check that the request was made before checking table status. From testing I can say this does happen in very specific scenarios. So far I was only able to repro this by doing a multiple table GSI update where one of the tables had a GSI removed prior and on the next deployment a GSI was added back with same name.